x/observer/keeper/keeper_block_header.go

```diff
@@ -0,0 +1,80 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/zeta-chain/zetacore/x/observer/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// SetNodeAccount set a specific nodeAccount in the store from its index
+func (k Keeper) SetBlockHeader(ctx sdk.Context, header types.BlockHeader) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.BlockHeaderKey))
+	b := k.cdc.MustMarshal(&header)
+	store.Set(header.Hash, b)
```

Comment:
why not add the chainID to the key ?

Reply:
that's a good idea

Reply:
Is it necessary? The `blockHash` should already be unique per se since the network ID is part of block hash calculation.

Reply:
That's my assumption as well. 
Adding chain-id is doable, will complicate things a bit, and the only benefit is that we can more efficiently query all blockheaders with given a chain-id.

Reply:
>Adding chain-id is doable, will complicate things a bit, and the only benefit is that we can more efficiently query all blockheaders with given a chain-id.

We would not be able to use pagination efficiently since the store is ordered in alphanumerical order.